### PR TITLE
feat: add qrscanner for addresses and invoices

### PR DESCRIPTION
**Pull request by dni**
Closes https://github.com/BoltzExchange/boltz-web-app/issues/300

qrcode scanner :)

- BREAKING: camera needs https to be working so i added mkcert plugin for vite, works great but from now on we need `https://localhost:5173/`

### CODE REVIEW

**Comment by kilrau** [comment]
- remove scan qr code button if not mobile
- when I do a reverse swap to mainchain but scan a liquid address qr it doesn't paste nor give me an error

**Review by michael1011 — CHANGES_REQUESTED** [review]
Scanned onchain addresses are set but not validated

File: src/Create.jsx [hunk 1]
```diff
@@ -291,6 +292,7 @@ const Create = () => {
                 <InvoiceInput />
                 <hr />
             </Show>
+            <QrScan />
```
[the remark this answers is not in the document]
Comment (dni) [reply src/Create.jsx]: done

**Review by michael1011 — CHANGES_REQUESTED** [review]
Clicking the image to close the QR code scanner without actually scanning a QR code is unintuitive. Keep the behaviour but put an X icon in some corner

**Comment by dni** [comment]
> Clicking the image to close the QR code scanner without actually scanning a QR code is unintuitive. Keep the behaviour but put an X icon in some corner

fixed

**Review by kilrau — CHANGES_REQUESTED** [review]
Can you rebase this branch? Looks like this has an address validation bug

image

**Comment by michael1011** [comment]
> Can you rebase this branch? Looks like this has an address validation bug

are you on the right network?

**Pull request merged**